akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -250,6 +250,11 @@ object Source {
    * The actor will be stopped when the stream is completed, failed or canceled from downstream,
    * i.e. you can watch it to get notified when that happens.
    *
+   * An async boundary is automatically created after the returned actorRef source, so downstream
+   * operations will not occur in the same thread as the Actor ingesting messages. If you select too
```

Comment:
Perhaps the warning can be rewritten to be less detailed. But I think it should at least warn users against setting a small value.